pkg/integrations/v2/controller.go

```diff
@@ -198,6 +200,23 @@ func (c *controller) UpdateController(cfg controllerConfig, globals Globals) err
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
+	var returnError error
+	// Check to ensure no duplicate singleton exist
+	singletonCheck := make(map[string]interface{})
```

Comment:
Use `struct{}` since it has no size (vs an `interface{}` which is always a fat pointer to something, even if it's nil)

```suggestion
	// Check to ensure no duplicate singleton exist
	singletonCheck := make(map[string]struct{})
```